src/generators/typescript/constrainer/EnumConstrainer.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { ConstrainedEnumModel, EnumModel } from '../../../models';
+import { NO_NUMBER_START_CHAR, NO_DUPLICATE_ENUM_KEYS, NO_EMPTY_VALUE, NO_RESERVED_KEYWORDS} from '../../../helpers/Constraints';
+import { FormatHelpers } from '../../../helpers';
+import { isReservedTypeScriptKeyword } from '../Constants';
+
+export type ModelEnumKeyConstraints = {
+  NO_SPECIAL_CHAR: (value: string) => string;
+  NO_NUMBER_START_CHAR: (value: string) => string;
+  NO_DUPLICATE_KEYS: (constrainedEnumModel: ConstrainedEnumModel, enumModel: EnumModel, value: string, namingFormatter: (value: string) => string) => string;
```

Comment:
Maybe it would be better to pass it as an object and not as a list of arguments, like:

```suggestion
  NO_DUPLICATE_KEYS: ({ constrainedEnumModel: ConstrainedEnumModel, enumModel: EnumModel, value: string, namingFormatter: (value: string) => string }) => string;
```

Here and also in another constraints.

Reply:
Yea, I will rework them once all the constrainers are in place, and make it prettier 😄